### PR TITLE
NLog integration

### PR DESCRIPTION
**Pull request by aholovko**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -131,17 +134,17 @@ There are multiple ways to set these values:
 
   ```csharp
   var config = new AirbrakeConfig {
-      ProjectId = "127348",
-      ProjectKey = "e2046ca6e4e9214b24ad252e3c99a0f6"
+      ProjectId = "113743",
+      ProjectKey = "81bbff95d52f8856c770bb39e827f3f6"
```
Comment (kyrylo) [inline README.md]: I'm not sure how this change is related to the NLog integration (and all other changes related to key/id). We probably want to revert these.

File: README.md [hunk 2]
```diff
@@ -370,9 +373,11 @@ airbrake.AddFilter(notice =>
         return null;
 
     // clear environment variables with "token"-related keys
-    foreach (var key in notice.EnvironmentVars.Keys)
-        if (key.Contains("token"))
+    new List<string>(notice.EnvironmentVars.Keys).ForEach(key =>
```
Comment (kyrylo) [inline README.md]: I don't see how this is relevant to the NLog integration. If the example is incorrect, it's probably better to open a separate PR.

File: README.md [hunk 3]
```diff
@@ -434,29 +439,29 @@ ASP.NET Integration
 1. Install the `Sharpbrake.Http.Module` package from NuGet (you can use "Package
    Manager Console" from Visual Studio):
 
-  ```
-  PM> Install-Package Sharpbrake.Http.Module
-  ```
+   ```
+   PM> Install-Package Sharpbrake.Http.Module
+   ```
```
Comment (kyrylo) [inline README.md]: Is this change intentional (and other changed related to indentation)? I don't see the reason. This probably doesn't help with the NLog integration.

File: README.md [hunk 4]
```diff
@@ -469,37 +474,189 @@ ASP.NET Integration
...
+     <logger name="*" minlevel="Error" writeTo="airbrake" />
+     <!-- other rules -->
+   </rules>
+   ```
+
+   Note that both `projectId` and `projectKey` are required parameters. You can
+   set any configuration option supported by the Airbrake client in the declarative
+   way ([how to configure](#configuration)).
+
+   In the case if you need programmatically access `Notifier` (e.g. for setting
```
Comment (kyrylo) [inline README.md]: `In the case if you need programmatically` -> ``When you need to access `Notifier` programmatically``

File: src/Sharpbrake.NLog.Web/AirbrakeTarget.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#if NET452
+        /// <summary>
+        /// Gets the current HTTP context.
+        /// </summary>
+        protected override IHttpContext GetHttpContext()
+        {
+            var httpContext = System.Web.HttpContext.Current;
+            return httpContext == null ? null : new AspNetHttpContext(httpContext);
+        }
+#elif NETSTANDARD1_4
```
Comment (kyrylo) [inline src/Sharpbrake.NLog.Web/AirbrakeTarget.cs]: Is it a new platform?

Comment (aholovko) [inline src/Sharpbrake.NLog.Web/AirbrakeTarget.cs]: Nope, for simplicity, this is what is called ".NET Core"

File: README.md (hunk 3 above)
Comment (aholovko) [inline README.md]: They are intentional, the previous indentation doesn't indent properly code snippets in relation to explanations.

File: README.md (hunk 4 above)
Comment (aholovko) [inline README.md]: Sounds good

File: README.md (hunk 3 above)
Comment (kyrylo) [inline README.md]: Can we move this to a different PR?

File: README.md (hunk 2 above)
Comment (aholovko) [inline README.md]: Example is incorrect. Would switching title from "Add documentation on NLog integration" to "Add documentation on NLog integration. Cleanup README" or something like that work here? Or using separate PRs are still the better way?

**Comment by aholovko** [comment]
@kyrylo PTAL

File: README.md [hunk 6]
```diff
@@ -501,6 +504,158 @@ ASP.NET Integration
...
+       });
+   
+       return notice;
+   });
+   ```
+
+### Airbrake NLog target for ASP.NET
+
+Airbrake NLog target for ASP.NET differs from the regular one in the additional
+support for the HTTP context.
```
Comment (kyrylo) [inline README.md]: This will sound a bit better:

> Airbrake NLog target for ASP.NET is almost identical to the standard target with the exception of support for HTTP context.

File: README.md [hunk 7]
```diff
@@ -501,6 +504,158 @@ ASP.NET Integration
...
+   ```
+
+   ```xml
+   <rules>
+     <logger name="*" minlevel="Error" writeTo="airbrake" />
+     <!-- other rules -->
+   </rules>
+   ```
+
+4. In the case if you are setting up the target for ASP.NET Core, extra step is
```
Comment (kyrylo) [inline README.md]: > To get reporting of HTTP context properties for ASP.NET Core you need to call the following code in the `Configure` method in `Startup.cs`:

File: README.md [hunk 8]
```diff
@@ -501,6 +504,158 @@ ASP.NET Integration
...
+   needed to get reporting of HTTP context properties. In the method `Configure`
+   of `Startup.cs` please call the next method:
+
+   ```csharp
+   app.ConfigureAirbrakeTarget();
+   ```
+
+   Here you have also access to the `Notifier` object that can be used for the
+   additional configuration from your code:
```
Comment (kyrylo) [inline README.md]: > The `Notifier` object can be accessed from here, too. It can be used for additional configuration:

File: src/Sharpbrake.NLog.Web/AirbrakeTarget.cs (hunk 5 above)
Comment (kyrylo) [inline src/Sharpbrake.NLog.Web/AirbrakeTarget.cs]: I remember it used to be `NETCORE`. Did something change?

**Comment by aholovko** [comment]
In past with old project format I have defined it manually in that way, while the new one has predefined consts.

File: README.md (hunk 7 above)
Comment (aholovko) [inline README.md]: +1

File: README.md (hunk 8 above)
Comment (aholovko) [inline README.md]: Does the word "too" here imply that there is another place where the `Notifier` object can be accessed? Or it means "in addition to [setting NLog, some framework features, etc.] you can access Notifier as well"?

Comment (kyrylo) [inline README.md]: The former.

File: README.md (hunk 6 above)
Comment (aholovko) [inline README.md]: That "is almost identical" is a little bit vague. What about:

> With Airbrake NLog target for ASP.NET you get reporting of HTTP context properties in web applications.

File: README.md (hunk 8 above)
Comment (aholovko) [inline README.md]: As for me, this one looks more accurate:

> The `Notifier` object can be accessed from here for the additional configuration:

With "too" it looks like we were talked about one place of accessing Notifier and this is another one.

Comment (kyrylo) [inline README.md]: Okay, no objections here.
